go/obscuronode/enclave/enclave.go

```diff
@@ -256,23 +259,23 @@ func (e *enclaveImpl) RoundWinner(parent obscurocommon.L2RootHash) (nodecommon.E
 
 	parentState := e.storage.FetchRollupState(head.Hash())
 	// determine the winner of the round
-	winnerRollup, s := findRoundWinner(usefulRollups, head, parentState, e.storage, e.blockResolver)
+	winnerRollup, s, hasTxs := e.findRoundWinner(usefulRollups, head, parentState)
 
 	e.storage.SetRollupState(winnerRollup.Hash(), s)
 	go e.notifySpeculative(winnerRollup)
 
 	// we are the winner
-	if winnerRollup.Header.Agg == e.node {
+	// and the rollup has either transfer or deposit transactions to publish
+	if winnerRollup.Header.Agg == e.node && hasTxs {
 		v := winnerRollup.Proof(e.blockResolver)
 		w := e.storage.ParentRollup(winnerRollup)
-		log.Log(fmt.Sprintf(">   Agg%d: create rollup=r_%d(%d)[r_%d]{proof=b_%d}. Txs: %v. State=%v.",
+		log.Log(fmt.Sprintf(">   Agg%d: publish rollup=r_%d(%d)[r_%d]{proof=b_%d}. Txs: %d. ",
```

Comment:
why is the state going away?

Reply:
Removed it because previously it was a string and was super huge. 
Adding it back ! :)